tests_and_analysis/test/euphonic_test/test_force_constants_object_creation.py

```diff
@@ -0,0 +1,282 @@
+        with pytest.raises(expected_exception):
+            ForceConstants(*faulty_args)
+
+    @pytest.fixture(params=[
+        (
+            'LZO', 'La2Zr2O7.castep_bin',
+            np.array([
+                [1.26492555e-01, -2.31204635e-31, -1.16997352e-13],
+                [-2.31204635e-31, 1.26492555e-01, 3.15544362e-30],
+                [-1.16997352e-13, 1.05181454e-30, 1.26492555e-01]
+            ]) * ureg('hartree/bohr**2'),
+            [3, 30, 33, 15, 18],
+            np.array([
+                [-8.32394989e-04, -2.03285211e-03, 3.55359333e-04],
+                [-2.22156212e-03, -6.29315975e-04, 1.21568713e-03],
+                [7.33617499e-05, 1.16282999e-03, 5.22410338e-05]
+            ]) * ureg('hartree/bohr**2')
+        ),
+        (
+            'graphite', 'graphite.castep_bin',
+            np.array([
+                [6.35111387e-01, 2.76471554e-18, 0.00000000e+00],
+                [2.05998413e-18, 6.35111387e-01, 0.00000000e+00],
+                [0.00000000e+00, 0.00000000e+00, 1.52513691e-01]
+            ]) * ureg('hartree/bohr**2'),
+            [10, 6, 9, 9, 12],
+            np.array([
+                [-8.16784177e-05, -1.31832252e-05, -1.11904290e-05],
+                [-1.31832252e-05, 5.42461675e-05, 3.73913780e-06],
+                [-1.11904290e-05, 3.73913780e-06, -2.99013850e-05]
+            ]) * ureg('hartree/bohr**2')
+        ),
+        (
+            'quartz', 'quartz.castep_bin',
+            np.array([
+                [0.22324308, 0.29855096, 0.31240272],
+                [0.29855096, 0.43968813, 0.34437270],
+                [0.31240272, 0.34437270, 0.33448280]
+            ]) * ureg('hartree/bohr**2'),
+            [4, 6, 9, 21, 24],
+            np.array([
+                [6.37874988e-05, -2.73116279e-04, -1.73624413e-04],
+                [1.99404810e-08, 4.40511248e-04, 2.33322837e-04],
+                [-1.85830030e-05, -9.05395392e-05, 2.17526152e-05]
+            ]) * ureg('hartree/bohr**2')
+        )
+    ])
+    def fc_mat(self, request):
+        directory, filename, fc_mat_cell0_i0_j0, celln, fc_mat_celln = \
+            request.param
+        path = os.path.join(get_data_path(), 'force_constants', directory)
+        expected_data = ExpectedData(
+            fc_mat_cell0_i0_j0=fc_mat_cell0_i0_j0,
+            fc_mat_celln=fc_mat_celln
+        )
+        filepath = os.path.join(path, filename)
+        fc = ForceConstants.from_castep(filepath)
+        return fc, celln, expected_data
+
+    def test_fc_mat_cells(self, fc_mat):
+        fc, celln, expected_data = fc_mat
+        npt.assert_allclose(
+            fc.force_constants[0, 0:3, 0:3].magnitude,
+            expected_data.fc_mat_cell0_i0_j0.magnitude
+        )
+        npt.assert_allclose(
+            fc.force_constants[
+                celln[0], celln[1]:celln[2], celln[3]:celln[4]
+            ].magnitude,
+            expected_data.fc_mat_celln.magnitude
+        )
```

Comment:
I don't think we need this part anymore (which tests a small part of the force constants matrix) as we're testing the entire matrix already in `check_force_constant_attrs`

Reply:
Ok, removed